### ISSUE TITLE
[Dropdown] Accessibility of dropdown items with arrow keys broken in 5.1

### ISSUE THREAD

**Issue by nsulzberger** [Esri/calcite-components#14467]
**monday.com sync:** #12071469073

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

N/A, Dropdown

### Summary

According to the documentation, the dropdown items can be accessed with the arrow keys, see: 
https://developers.arcgis.com/calcite-design-system/components/dropdown/#accessibility

We are using the dropdown component without an element in the trigger slot. With Calcite 5.1 the arrow keys cannot be used anymore to access the items. With 5.0 it was working.

### Actual Behavior

Using the arrow keys when the focus is on the first item of the dropdown has no effect.

### Expected Behavior

 The dropdown items can be accessed with the arrow keys.

### Reproduction Sample

https://codepen.io/nsulzberger_esri/pen/PwbmbNq?editors=100

You can also repro with a calcite-menu and calcite-menu-item as a trigger:
https://codepen.io/nsulzberger_esri/pen/RNoVEdE?editors=1100

### Reproduction Steps

1. open the sample
2. click on "Natural Places" to make sure that your focus is on the element
3. use the arrow keys -> nothing happens
4. change the code, switch to Calcite 5.0 -> the same steps are working

### Reproduction Version

5.1

### Regression?

5.0

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Scene Viewer

**Comment by driskull** [comment]
I'm not sure that the first example is a valid use case. I believe that there needs to be either a trigger or referenceElement.

The second example doesn't work because the event is prevented (cancelled)

**Comment by DitwanP** [comment]
Spike to confirm behavior for next steps based on Matt's comment above

**Comment by nsulzberger** [comment]
We managed to change the UI a bit in Scene Viewer and have added an action as a trigger. So we don't need a fix for it. But I wonder if we were the only ones using a dropdown like this.

**Comment by DitwanP** [comment]
Although the original issue poster no longer need this fix, after looking into it, to me it seems like a valid bug. 

Even with a trigger, if the dropdown has the `open` property set, when the component first loads, you can't navigate with the arrows. 

However, you can navigate as expected once you close and reopen via keyboard or mouse.

https://codepen.io/Ditwan-Price/pen/ByQwONZ

Looks like it first showed up in `5.1.0-next.14` with this [PR](https://github.com/Esri/calcite-design-system/pull/14079) that was meant to change some related behavior, I'm not sure if this was an intended outcome. 

cc @driskull to confirm.

**Comment by driskull** [comment]
@DitwanP that codepen link doesn't work.

**Comment by DitwanP** [comment]
> [@DitwanP](https://github.com/DitwanP) that codepen link doesn't work.

@driskull Oops, here you go: https://codepen.io/Ditwan-Price/pen/ByQwONZ

**Comment by driskull** [comment]
> Even with a trigger, if the dropdown has the open property set, when the component first loads, you can't navigate with the arrows.

@DitwanP it does work if you have the trigger focused. I wouldn't expect it to work if the trigger element isn't focused as just an arrow key on the body shouldn't move the active/selected dropdown.

**Comment by DitwanP** [comment]
@driskull Yeah that makes sense, but what about a case where there isn't a trigger, are we expecting that there should always be one? Or that the developer should set the focus to the dropdown trigger if it loads already opened?

**Comment by driskull** [comment]
> what about a case where there isn't a trigger, are we expecting that there should always be one? Or that the developer should set the focus to the dropdown trigger if it loads already opened?

I'm not sure "no trigger" is a valid use case.

> are we expecting that there should always be one

I think so. The trigger/referenceElement is how the dropdown is positioned correctly

**Comment by DitwanP** [comment]
Spike findings: The reported behavior occurs because the component is not configured with a trigger or referenceElement. Based on Matt’s comment above and confirmation from design that a Dropdown is expected to always have a trigger/referenceElement, we should close this as not planned.

**Comment by geospatialem** [comment]
Closing the above as unplanned per the discussions above, refer to https://github.com/Esri/calcite-design-system/issues/14467#issuecomment-4510251134 and https://github.com/Esri/calcite-design-system/issues/14467#issuecomment-4549242273.
